modules/apps/collaboration/announcements/announcements-uad/src/main/java/com/liferay/announcements/uad/exporter/BaseAnnouncementsUADEntityExporter.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.announcements.uad.exporter;
+
+import com.liferay.announcements.constants.AnnouncementsPortletKeys;
+import com.liferay.document.library.kernel.exception.NoSuchFolderException;
+import com.liferay.document.library.kernel.model.DLFolderConstants;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.GroupConstants;
+import com.liferay.portal.kernel.model.Repository;
+import com.liferay.portal.kernel.portletfilerepository.PortletFileRepositoryUtil;
+import com.liferay.portal.kernel.repository.model.Folder;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.user.associated.data.exporter.BaseUADEntityExporter;
+
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Drew Brokke
+ */
+public abstract class BaseAnnouncementsUADEntityExporter
+	extends BaseUADEntityExporter {
+
```

Comment:
Looks like we have a lot of duplicate code in the exporter classes. We can just extract the code to the `BaseUADEntityExporter` instead of creating a new base class since these code will also be used by other UAD entity exporters.

Reply:
I agree; I think a lot of this code will be reusable by `Exporter` classes in other applications.

Reply:
I would _love_ to find a way to make those `verify` methods generic.  I attempted it briefly using generics and it got pretty hairy. I think we can do it though.